pkg/executor/openstack.go

```diff
@@ -0,0 +1,593 @@
+	"github.com/gophercloud/gophercloud/openstack/compute/v2/extensions/keypairs"
+	"github.com/gophercloud/gophercloud/openstack/compute/v2/extensions/startstop"
+	"github.com/gophercloud/gophercloud/openstack/compute/v2/flavors"
+	"github.com/gophercloud/gophercloud/openstack/compute/v2/images"
+	"github.com/gophercloud/gophercloud/openstack/compute/v2/servers"
+	"github.com/intelsdi-x/swan/pkg/conf"
+	"github.com/intelsdi-x/swan/pkg/utils/uuid"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	executorName        = "Openstack executor"
+	executorLogPrefix   = executorName + ":"
+	taskHandleName      = "Openstack task handle"
+	taskHandleLogPrefix = taskHandleName + ":"
+	directoryPrefix 	= "openstack"
+)
+
+var (
+	flavorDiskFlag  = conf.NewIntFlag("flavor_disk", "Openstack flavor disk size [GB]", 10)
+	flavorRAMFlag   = conf.NewIntFlag("flavor_ram", "Openstack flavor RAM size [MB]", 1024)
+	flavorVCPUsFlag = conf.NewIntFlag("flavor_vcpus", "Openstack flavor VCPUs", 1)
+	imageFlag       = conf.NewStringFlag("image", "Name of image.", "cirros")
+	userFlag        = conf.NewStringFlag("username", "Username", "cirros")
+	sshKeyPathFlag  = conf.NewStringFlag("ssh_key", "SSH key path", "~/.ssh/id_rsa")
+	keypairName     = conf.NewStringFlag("os_keypair_name", "Openstack Keypair Name", "swan")
+	bootUpTimeOut   = conf.NewDurationFlag("vm_boot_up_timeout", "Virtual Machine boot up timeout", time.Second*30)
+)
+
+// DefaultOpenstackConfig creates default OpenStack config.
+func DefaultOpenstackConfig(auth gophercloud.AuthOptions) OpenstackConfig {
+	return OpenstackConfig{
+		Auth: auth,
+		Flavor: OpenstackFlavor{
+			Disk:  flavorDiskFlag.Value(),
+			RAM:   flavorRAMFlag.Value(),
+			VCPUs: flavorVCPUsFlag.Value(),
+		},
+		Image:      imageFlag.Value(),
+		User:       userFlag.Value(),
+		SSHKeyPath: sshKeyPathFlag.Value(),
+	}
+}
+
+// OpenstackFlavor defines OpenStack flavor data.
+type OpenstackFlavor struct {
+	Disk  int
+	RAM   int
+	VCPUs int
+}
+
+// OpenstackAuthConfig defines OpenStack authentication configuration data.
+type OpenstackAuthConfig struct {
+	Username   string
+	Password   string
+	TenantID   string
+	DomainName string
+	Endpoint   string
+}
+
+// OpenstackConfig defines OpenStack instance configuration data.
+type OpenstackConfig struct {
+	Auth                   gophercloud.AuthOptions
+	Flavor                 OpenstackFlavor
+	FlavorName             string
+	Image                  string
+	User                   string
+	SSHKeyPath             string
+	HypervisorInstanceName string
+}
+
+// Openstack defines OpenStack server configuration and client.
+type Openstack struct {
+	config *OpenstackConfig
+	client *gophercloud.ServiceClient
+}
+
+// NewOpenstack creates OpenStack executor.
+func NewOpenstack(config *OpenstackConfig) Executor {
+	return Openstack{config: config}
+}
+
+// String returns user-friendly name of executor.
+func (os Openstack) String() string {
```

Comment:
General remark to be applied for whole file: first you import ``os`` package from Golang stdlib and they you use ``os`` as a variable name for instances of an openstack executor. This is technically correct and code compiles but I started wondering why you decided to use ``os`` package to handle keys and flavors. General rule of thumb is: do not use same identifier for a variable and imported package.
I might have requested you to Wrap() some comments that should not have been wrapped because of this :(